experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -2,36 +2,44 @@
 # 1: Construct auxiliary base space
 ################################################################
 
+
 function _auxiliary_base_space(auxiliary_base_variable_names::Vector{String}, auxiliary_base_grading::Matrix{Int64}, d::Int)
+
+  # We now try to guess one toric base space with the desired grading and maximal dimension
+  charges = matrix(ZZ, auxiliary_base_grading)
+  variety = normal_toric_variety_from_glsm(charges)
+  G1 = free_abelian_group(ncols(charges))
+  G2 = free_abelian_group(nrows(charges))
+  set_attribute!(variety, :map_from_torusinvariant_weil_divisor_group_to_class_group, hom(G1, G2, transpose(charges)))
+  set_attribute!(variety, :class_group, G2)
+  set_attribute!(variety, :torusinvariant_weil_divisor_group, G1)
   
-  # Find candidate base spaces
-  candidates = normal_toric_varieties_from_glsm(matrix(ZZ, auxiliary_base_grading))
-  @req length(candidates) > 0 "Could not find a full star triangulation"
-  f = candidates[1]
-  @req dim(f) >= d "Cannot construct an auxiliary base space of the desired dimension"
+  # Check if dimensional requirement is met
+  @req dim(variety) >= d "Cannot construct an auxiliary base space of the desired dimension"
   
   # Construct one base space of desired dimension
-  fan_rays = matrix(ZZ, rays(f))
-  fan_rays = [vec([Int(a) for a in fan_rays[k,:]]) for k in 1:nrows(fan_rays)]
-  fan_max_cone_matrix = matrix(ZZ, cones(f))
-  new_max_cones = Vector{Int}[]
-  for k in 1:nrows(fan_max_cone_matrix)
-    indices = findall(x -> x == 1, vec(fan_max_cone_matrix[k,:]))
-    if length(indices) == d
-      push!(new_max_cones, indices)
+  auxiliary_base_space = variety
+  if dim(auxiliary_base_space) != d
+    integral_rays = matrix(ZZ, rays(variety))
+    integral_rays = [vec([Int(a) for a in integral_rays[k,:]]) for k in 1:nrows(integral_rays)]
+    all_cones = cones(variety)
+    fan_max_cone_matrix = matrix(ZZ, cones(variety))
+    new_max_cones = Vector{Int}[]
+    for k in 1:nrows(fan_max_cone_matrix)
+      cone = positive_hull(integral_rays[all_cones[k,:]])
+      if dim(cone) == d
+        indices = findall(x -> x == 1, vec(fan_max_cone_matrix[k,:]))
+        push!(new_max_cones, indices)
+      end
     end
+    auxiliary_base_space = normal_toric_variety(integral_rays, new_max_cones; non_redundant = true)
```

Comment:
This is the reason why I was reluctant to add `cones(*)` as opposed to `cones(*,Int)`, because then people forget about the latter and just use the former and use some weird expensive filters. Everytime you call `positive_hull` a new cone is assembled and you restart computations that are already in the fan.

Something like the following should work:
```suggestion
    new_max_cones = IncidenceMatrix(cones(variety, d))
    auxiliary_base_space = normal_toric_variety(integral_rays, new_max_cones; non_redundant = true)
```
In case this does not work immediately, here is an example:
```
julia> p2 = projective_space(NormalToricVariety, 2)
Normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor

julia> skel = normal_toric_variety(matrix(ZZ,rays(p2)), IncidenceMatrix(cones(p2, 1)))
Normal toric variety

julia> dim(skel)
1

julia> skel = normal_toric_variety(rays(p2), IncidenceMatrix(cones(p2, 1)))
Normal toric variety

julia> dim(skel)
1
```